src/ixdat/measurements.py

```diff
@@ -269,22 +286,203 @@ def data_cols(self):
+
+        t, v = self.get_t_and_v(series_name)
+        mask = v == value  # linter doesn't realize this is a np array
+        mask_prev = np.append(False, mask[:-1])
+        mask_next = np.append(mask[1:], False)
+        interval_starts_here = np.logical_and(
+            np.logical_not(mask_prev), mask
+        )  # True at [0] if mask[0] is True.
+        interval_ends_here = np.logical_and(
+            mask, np.logical_not(mask_next)
+        )  # True at [-1] if mask[-1] is True.
+        t_starts = list(t[interval_starts_here])
+        t_ends = list(t[interval_ends_here])
+        tspans = zip(t_starts, t_ends)
+        meas = None
+        for tspan in tspans:
+            if meas:
+                meas = meas + new_measurement.cut(tspan)
+            else:
+                meas = new_measurement.cut(tspan)
+        new_measurement = meas
+        return new_measurement
+
+    def select_values(self, *args, **kwargs):
+        """Return a new Measurement with the time(s) in the measurement meeting criteria
+
+        Any series can be selected for using the series name as a key-word. Arguments
+        can be single acceptable values or lists of acceptable values. In the latter
+        case, each acceptable value is selected for on its own and the resulting
+        measurements added together.
+        # FIXME: That is sloppy because it mutliplies the number of DataSeries
+            containing the same amount of data.
+        If no key-word is given, the series name is assumed to
+        be the default selector, which is named by self.sel_str. Multiple criteria are
+        applied sequentially, i.e. you get the intersection of satisfying parts.
+
+        Args:
+            args (tuple): Argument(s) given without key-word are understood as acceptable
+                value(s) for the default selector (that named by self.sel_str)
+            kwargs (dict): Each key-word arguments is understood as the name
+                of a series and its acceptable value(s).
+        """
+        if len(args) >= 1:
+            if not self.sel_str:
+                raise BuildError(
+                    f"{self} does not have a default selection string "
+                    f"(Measurement.sel_str), and so selection only works with kwargs."
+                )
+            if len(args) == 1:
+                args = args[0]
+            kwargs[self.sel_str] = args
+        new_measurement = self
+        for series_name, allowed_values in kwargs.items():
+            if not hasattr(allowed_values, "__iter__"):
+                allowed_values = [allowed_values]
+            meas = None
+            for value in allowed_values:
+                m = new_measurement.select_value(**{series_name: value})
+                if meas:
+                    meas = meas + m
+                else:
+                    meas = m
+            new_measurement = meas
+        return new_measurement
+
+    def select(self, *args, tspan=None, **kwargs):
```

Comment:
Not sure if I understand correctly what the select_x functions exactly do. Is it to return a measurement object that contains e.g. only the signal for M2 in a given tspan?

Reply:
No, the measurement it returns contains every `DataSeries` which has data corresponding to the time being selected for! Select just gives access to cutting and selecting in the same function.
You can say `select(tspan=[100, 200], file_number=1)`, for example.
Ideally you should be able to say e.g., `select(cycle=1, 0.5<potential<1)`, but that's not implemented yet.

Is this clear? Does something need to be added to the docstring or changed?